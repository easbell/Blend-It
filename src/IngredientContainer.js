import React, { Component } from 'react';
import Ingredient from './Ingredient.js';

export class IngredientContainer extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      categories: ['fruit', 'vegetables', 'bases', 'extras']
    }
  }

  gatherIngredients = () => {
    return this.state.categories.map(category => {
      return (
        <div key={category}>
          <h3>{category}</h3>
          { this.props.ingredients[category].map(ingredient => {
            return (
              <Ingredient 
                ingredient={ingredient} 
                key={ingredient} 
                chooseIngredients={this.props.chooseIngredients}
                removeIngredient={this.props.removeIngredient}
              />
            )
          }).slice(0, 6) }
        </div>)
    })
  }
  
  render() {
    return (
      <div className='ingredients-container'>
        {this.gatherIngredients()}
      </div>
    )
  }
}

export default IngredientContainer;