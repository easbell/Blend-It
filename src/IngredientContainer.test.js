import React from 'react';
import IngredientContainer from './IngredientContainer.js';
import { shallow } from 'enzyme';

const mockIngredients = {
  fruit: [
  "banana",
  "pineapple",
  "mango",
  "cherries",
  "blueberries",
  "strawberries",
  "peach"
  ],
  vegetables: [
  "spinach",
  "kale",
  "carrot",
  "power greens mix",
  "cucumber",
  "beet"
  ],
  bases: [
  "milk",
  "yogurt",
  "coconut water",
  "orange juice"
  ],
  extras: [
  "chia seeds",
  "ginger",
  "cinnamon",
  "protein powder",
  "honey",
  "almond butter",
  "vanilla extract",
  "mint leaves"
  ]
}

const chooseIngredientsMock = jest.fn();
const removeIngredientMock = jest.fn();

describe('IngredientContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <IngredientContainer 
        ingredients={mockIngredients} 
        chooseIngredients={chooseIngredientsMock}
        removeIngredient={removeIngredientMock} 
      />
    );
  }); 

  it('should match a snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should should have property default states', () => {
    expect(wrapper.state()).toEqual({categories: Object.keys(mockIngredients)})
  })

});

