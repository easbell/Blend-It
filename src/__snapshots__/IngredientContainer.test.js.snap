// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientContainer should match a snapshot with all data passed in 1`] = `
<div
  className="ingredients-container"
>
  <div
    key="fruit"
  >
    <h3>
      fruit
    </h3>
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="banana"
      key="banana"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="pineapple"
      key="pineapple"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="mango"
      key="mango"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="cherries"
      key="cherries"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="blueberries"
      key="blueberries"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="strawberries"
      key="strawberries"
      removeIngredient={[MockFunction]}
    />
  </div>
  <div
    key="vegetables"
  >
    <h3>
      vegetables
    </h3>
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="spinach"
      key="spinach"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="kale"
      key="kale"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="carrot"
      key="carrot"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="power greens mix"
      key="power greens mix"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="cucumber"
      key="cucumber"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="beet"
      key="beet"
      removeIngredient={[MockFunction]}
    />
  </div>
  <div
    key="bases"
  >
    <h3>
      bases
    </h3>
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="milk"
      key="milk"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="yogurt"
      key="yogurt"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="coconut water"
      key="coconut water"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="orange juice"
      key="orange juice"
      removeIngredient={[MockFunction]}
    />
  </div>
  <div
    key="extras"
  >
    <h3>
      extras
    </h3>
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="chia seeds"
      key="chia seeds"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="ginger"
      key="ginger"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="cinnamon"
      key="cinnamon"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="protein powder"
      key="protein powder"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="honey"
      key="honey"
      removeIngredient={[MockFunction]}
    />
    <Ingredient
      chooseIngredients={[MockFunction]}
      ingredient="almond butter"
      key="almond butter"
      removeIngredient={[MockFunction]}
    />
  </div>
</div>
`;
